apmproxy/receiver.go

```diff
@@ -91,6 +97,19 @@ func (c *Client) handleInfoRequest() (func(w http.ResponseWriter, r *http.Reques
 		// Don't update the status of the transport as it is possible that the extension
 		// is frozen while processing the request and context is canceled due to timeout.
 		c.logger.Errorf("Error querying version from the APM server: %v", err)
+
+		uErr := new(unexpectedStatusError)
+		if errors.As(err, uErr) {
+			status := int(*uErr)
+
+			// Proxy the status code to the agent.
+			w.WriteHeader(status)
+			return
+		}
```

Comment:
```suggestion
```

AFAICS, this code path is completely dead - since ModifyResponse was returned, err will never be unexpectedStatusError.